connector/datadogconnector/factory.go

```diff
@@ -35,7 +36,15 @@ func createDefaultConfig() component.Config {
 // defines the consumer type of the connector
 // we want to consume traces and export metrics therefore define nextConsumer as metrics, consumer is the next component in the pipeline
 func createTracesToMetricsConnector(_ context.Context, params connector.CreateSettings, cfg component.Config, nextConsumer consumer.Metrics) (connector.Traces, error) {
-	c, err := newConnector(params.Logger, cfg, nextConsumer)
+	c, err := newConnector(params.Logger, cfg, nextConsumer, nil)
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func createTracesToTracesConnector(_ context.Context, params connector.CreateSettings, cfg component.Config, nextConsumer consumer.Traces) (connector.Traces, error) {
+	c, err := newConnector(params.Logger, cfg, nil, nextConsumer)
```

Comment:
Should the traces-to-traces and traces-to-metrics instances be the same connector? Right now they are two different independent instances of the connector. I think there is no shared state so it shouldn't matter but I want to double check

Reply:
You mean use a singleton instance for connector? That doesn't seem a common practice for collector components AFAIS.

Otherwise I don't think traces-to-traces and traces-to-metrics connectors can be the same instance
- traces-to-traces only has trace consumer, no metrics consumer
- traces-to-metrics only has metrics consumer, no traces consumer

Reply:
> You mean use a singleton instance for connector? That doesn't seem a common practice for collector components AFAIS.

I meant a singleton per config, the same way it is done for e.g. the OTLP receiver:
https://github.com/open-telemetry/opentelemetry-collector/blob/287b98f6973fd6baa278150b9fca8c83abea0af4/receiver/otlpreceiver/factory.go#L71-L77

> Otherwise I don't think traces-to-traces and traces-to-metrics connectors can be the same instance

The worry is that we may need to share state via the trace Agent. I don't think it's a concern today though, right? (as in, no shared state in the trace agent that would be relevant). That's what I wanted to confirm